src/Mvc/Mvc.Core/src/ModelBinding/FormValueProviderFactory.cs

```diff
@@ -33,9 +36,20 @@ public Task CreateValueProviderAsync(ValueProviderFactoryContext context)
         private static async Task AddValueProviderAsync(ValueProviderFactoryContext context)
         {
             var request = context.ActionContext.HttpContext.Request;
+            IFormCollection form;
+            try
```

Comment:
Don't catch the exception, use `request .HasFormContentType `

Reply:
I would like not to catch exception. As far as I remember `HasFormContentType` returns true for "multipart/form-data" without boundary. Also `ReadFormAsync` throws `InvalidDataException` in many other cases where it has problem with parsing form.

Reply:
I also thought about introducing new method var errors = request.TryReadFormAsync(out form) - would that be a way?

Reply:
@Tratcher

Reply:
TryReadFormAsync would require a whole new call stack / form parsing implementation. Some of those exceptions come from the very bottom. Since forms are rarely invalid I think you're better off catching the exception.

Reply:
Agreed. After looking at this catching might be best